ibm/service/power/resource_ibm_pi_instance.go

```diff
@@ -840,20 +847,23 @@ func resourceIBMPIInstanceDelete(ctx context.Context, d *schema.ResourceData, me
 		return diag.FromErr(err)
 	}
 
-	cloudInstanceID, instanceID, err := splitID(d.Id())
+	idArr, err := flex.IdParts(d.Id())
 	if err != nil {
 		return diag.FromErr(err)
 	}
 
-	client := st.NewIBMPIInstanceClient(ctx, sess, cloudInstanceID)
-	err = client.Delete(instanceID)
-	if err != nil {
-		return diag.FromErr(err)
-	}
+	cloudInstanceID := idArr[0]
+	for _, instanceID := range idArr[1:] {
+		client := st.NewIBMPIInstanceClient(ctx, sess, cloudInstanceID)
+		err = client.Delete(instanceID)
+		if err != nil {
+			return diag.FromErr(err)
+		}
 
-	_, err = isWaitForPIInstanceDeleted(ctx, client, instanceID)
-	if err != nil {
-		return diag.FromErr(err)
+		_, err = isWaitForPIInstanceDeleted(ctx, client, instanceID)
```

Comment:
If instead of delete-wait and loop again, can we have two loops? One loop will delete all the instances and the other loop will check the deletion status. This will save lots of time.

Reply:
Made changes.